maven-scm-api/src/main/java/org/apache/maven/scm/command/changelog/AbstractChangeLogCommand.java

```diff
@@ -50,11 +50,20 @@ protected abstract ChangeLogScmResult executeChangeLogCommand( ScmProviderReposi
     protected ChangeLogScmResult executeChangeLogCommand( ScmProviderRepository repository, ScmFileSet fileSet,
                                                           ScmVersion startVersion, ScmVersion endVersion,
                                                           String datePattern )
-        throws ScmException
+            throws ScmException
     {
         throw new ScmException( "Unsupported method for this provider." );
     }
 
+    protected ChangeLogScmResult executeChangeLogCommand( ScmProviderRepository repository, ScmFileSet fileSet,
+                                                          ScmVersion endVersion,
+                                                          String datePattern, boolean startFromRoot )
+            throws ScmException
+    {
+        // by default let's just delegate to executeChangeLogCommand with both start and end versions
+        return executeChangeLogCommand( repository, fileSet, null, endVersion, datePattern );
```

Comment:
What is the purpose of this entire method if `startFromRoot` is not passed?

Reply:
was to have a default non throwing exception behavior ; i'll use `throw new ScmException( "Unsupported method for this provider." );` instead

Reply:
Just throw `UnsupportedOperationException`.